packages/twenty-front/src/modules/settings/profile/components/DateTimeSettingsTimeFormatSelect.tsx

```diff
@@ -0,0 +1,51 @@
+import { formatInTimeZone } from 'date-fns-tz';
+
+import { Select } from '@/ui/input/components/Select';
+import { TimeFormat } from '@/workspace-member/constants/TimeFormat';
+import { detectTimeZone } from '@/workspace-member/utils/detectTimeZone';
+
+type DateTimeSettingsTimeFormatSelectProps = {
+  value: TimeFormat;
+  onChange: (nextValue: TimeFormat) => void;
+  timeZone: string;
+};
+
+export const DateTimeSettingsTimeFormatSelect = ({
+  onChange,
+  timeZone,
+  value,
+}: DateTimeSettingsTimeFormatSelectProps) => {
+  const setTimeZone = timeZone === 'system' ? detectTimeZone() : timeZone;
+  return (
+    <Select
+      dropdownId="datetime-settings-time-format"
+      dropdownWidth={218}
+      label="Time format"
+      fullWidth
+      value={value}
+      options={[
+        {
+          label: 'System settings',
+          value: TimeFormat.SYSTEM,
+        },
+        {
+          label: `24h (${formatInTimeZone(
+            Date.now(),
+            setTimeZone,
+            TimeFormat.MILITARY,
+          )})`,
+          value: TimeFormat.MILITARY,
+        },
+        {
+          label: `12h (${formatInTimeZone(
+            Date.now(),
+            setTimeZone,
+            TimeFormat.STANDARD,
+          )})`,
+          value: TimeFormat.STANDARD,
```

Comment:
How come 12 hours is "standard" 😁. We use 24hours in France ;)

Reply:
haha, I agree the naming is a little confusing here, its the "format" for time here which is "standard" if that makes sense. Whichever timezone is detected it will set it automatically to that "format" for that respective timezone